src/main/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporter.java

```diff
@@ -0,0 +1,341 @@
+import org.prebid.server.analytics.reporter.greenbids.model.CommonMessage;
+import org.prebid.server.analytics.reporter.greenbids.model.ExtBanner;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsAnalyticsProperties;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsBidder;
+import org.prebid.server.analytics.reporter.greenbids.model.HttpUtil;
+import org.prebid.server.analytics.reporter.greenbids.model.MediaTypes;
+import org.prebid.server.auction.model.AuctionContext;
+import org.prebid.server.auction.model.BidRejectionTracker;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.EncodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.request.ExtImpPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtStoredRequest;
+import org.prebid.server.proto.openrtb.ext.response.seatnonbid.NonBid;
+import org.prebid.server.proto.openrtb.ext.response.seatnonbid.SeatNonBid;
+import org.prebid.server.vertx.httpclient.HttpClient;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GreenbidsAnalyticsReporter implements AnalyticsReporter {
+
+    private final GreenbidsAnalyticsProperties greenbidsAnalyticsProperties;
+    private final JacksonMapper jacksonMapper;
+    private final HttpClient httpClient;
+    private static final int MAX_16_BIT_INTEGER = 0xFFFF;
+    private static final int RANGE_16_BIT_INTEGER_DIVISION_BASIS = 0x10000;
+
+    public GreenbidsAnalyticsReporter(
+            GreenbidsAnalyticsProperties greenbidsAnalyticsProperties,
+            JacksonMapper jacksonMapper,
+            HttpClient httpClient) {
+        this.greenbidsAnalyticsProperties = Objects.requireNonNull(greenbidsAnalyticsProperties);
+        this.jacksonMapper = Objects.requireNonNull(jacksonMapper);
+        this.httpClient = Objects.requireNonNull(httpClient);
+    }
+
+    @Override
+    public <T> Future<Void> processEvent(T event) {
+        final AuctionContext greenbidsAuctionContext;
+        final BidResponse greenbidsBidResponse;
+
+        if (event instanceof AmpEvent ampEvent) {
+            greenbidsAuctionContext = ampEvent.getAuctionContext();
+            greenbidsBidResponse = ampEvent.getBidResponse();
+        } else if (event instanceof AuctionEvent auctionEvent) {
+            greenbidsAuctionContext = auctionEvent.getAuctionContext();
+            greenbidsBidResponse = auctionEvent.getBidResponse();
+        } else {
+            greenbidsAuctionContext = null;
+            greenbidsBidResponse = null;
+        }
```

Comment:
```
        AuctionContext greenbidsAuctionContext = null;
        BidResponse greenbidsBidResponse = null;

        if (event instanceof AmpEvent ampEvent) {
            greenbidsAuctionContext = ampEvent.getAuctionContext();
            greenbidsBidResponse = ampEvent.getBidResponse();
        } else if (event instanceof AuctionEvent auctionEvent) {
            greenbidsAuctionContext = auctionEvent.getAuctionContext();
            greenbidsBidResponse = auctionEvent.getBidResponse();
        }
```

Also, I believe the naming can be simplified to `auctionContext` and `bidResponse`